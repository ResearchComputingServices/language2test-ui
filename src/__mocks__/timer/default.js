export default {
    tick: 0,
    totalTick: 0,
    timestamp: null,
    timerActive: false,
};
