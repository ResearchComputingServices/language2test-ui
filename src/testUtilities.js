/* eslint-disable import/prefer-default-export */
/* eslint-disable prefer-const */
/* eslint-disable react/prop-types */
/* eslint-disable react/no-children-prop */
import React from 'react';
import { shallow, mount as mountComponent } from 'enzyme';
import { Router } from 'react-router-dom';
import toJSON from 'enzyme-to-json';
import { Provider } from 'react-redux';
import configureMockStore from 'redux-mock-store';
import _ from 'lodash';
import historyService from './services/HistoryService';
import { reducers } from './redux/slices';

const mockStore = configureMockStore({ reducers });

function WrapperComponent({ propsFunc, children, Component }) {
    const props = propsFunc();
    return (
        <Component
            children={children}
            {...props}
        />
    );
}

export function createTest(Component, opts = {}, callback) {
    let { props, children, mount, state } = opts;
    if (_.isNil(props)) props = {};
    if (_.isNil(children)) children = null;
    if (mount) {
        return () => {
            state = state || {};
            const mockMath = Object.create(global.Math);
            mockMath.random = () => 0.5;
            global.Math = mockMath;
            const mountedTree = mountComponent((
                <Router history={historyService.getHistory()}>
                    <Provider store={mockStore({ ...state })}>
                        <Component {...props} >
                            {children}
                        </Component>
                    </Provider>
                </Router>));
            _.isFunction(callback) && callback(mountedTree);
        };
    }
    if (_.isFunction(props)) {
        return () => {
            const shallowTree = shallow(
                <WrapperComponent
                    Component={Component}
                    propsFunc={props}
                >
                    {children}
                </WrapperComponent>,
            ).dive();
            _.isFunction(callback) && callback(shallowTree);
        };
    }
    return () => {
        const shallowTree = shallow((
            <Component {...props} >
                {children}
            </Component>
        ));
        _.isFunction(callback) && callback(shallowTree);
    };
}

export function createSnapshotTest(Component, opts = {}) {
    return createTest(Component, opts, tree => expect(toJSON(tree)).toMatchSnapshot());
}
