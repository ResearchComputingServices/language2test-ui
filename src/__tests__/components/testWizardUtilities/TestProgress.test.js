import Component from '../../../components/testWizardUtilities/TestProgress';
import timer from '../../../__mocks__/timer/default';
import { createSnapshotTest } from '../../../testUtilities';

describe('TestProgress', () => {
    it('should match snapshot', createSnapshotTest(Component, {
        props: {
            current: 0,
            total: 100,
            timer: 120,
            tick: 30,
            totalTick: 130,
        },
        state: { timer },
        mount: true,
    }));
});
