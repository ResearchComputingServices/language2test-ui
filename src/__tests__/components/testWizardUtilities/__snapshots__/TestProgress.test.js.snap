// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TestProgress should match snapshot 1`] = `
<Router
  history={
    Object {
      "action": "POP",
      "block": [Function],
      "createHref": [Function],
      "go": [Function],
      "goBack": [Function],
      "goForward": [Function],
      "length": 1,
      "listen": [Function],
      "location": Object {
        "hash": "",
        "pathname": "/",
        "search": "",
        "state": undefined,
      },
      "push": [Function],
      "replace": [Function],
    }
  }
>
  <Provider
    store={
      Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      }
    }
  >
    <TestProgress
      current={0}
      tick={30}
      timer={120}
      total={100}
      totalTick={130}
    >
      <div
        className="test-progress"
      >
        <div
          className="test-progress-header"
        >
          <h5>
            0 out of 100
          </h5>
          <h5>
            Time Remaining: 00:00
          </h5>
        </div>
        <WithStyles(ForwardRef(LinearProgress))
          className="test-progress-bar"
          color="primary"
          value={NaN}
          variant="determinate"
        >
          <ForwardRef(LinearProgress)
            className="test-progress-bar"
            classes={
              Object {
                "bar": "MuiLinearProgress-bar",
                "bar1Buffer": "MuiLinearProgress-bar1Buffer",
                "bar1Determinate": "MuiLinearProgress-bar1Determinate",
                "bar1Indeterminate": "MuiLinearProgress-bar1Indeterminate",
                "bar2Buffer": "MuiLinearProgress-bar2Buffer",
                "bar2Indeterminate": "MuiLinearProgress-bar2Indeterminate",
                "barColorPrimary": "MuiLinearProgress-barColorPrimary",
                "barColorSecondary": "MuiLinearProgress-barColorSecondary",
                "buffer": "MuiLinearProgress-buffer",
                "colorPrimary": "MuiLinearProgress-colorPrimary",
                "colorSecondary": "MuiLinearProgress-colorSecondary",
                "dashed": "MuiLinearProgress-dashed",
                "dashedColorPrimary": "MuiLinearProgress-dashedColorPrimary",
                "dashedColorSecondary": "MuiLinearProgress-dashedColorSecondary",
                "determinate": "MuiLinearProgress-determinate",
                "indeterminate": "MuiLinearProgress-indeterminate",
                "query": "MuiLinearProgress-query",
                "root": "MuiLinearProgress-root",
              }
            }
            color="primary"
            value={NaN}
            variant="determinate"
          >
            <div
              aria-valuemax={100}
              aria-valuemin={0}
              aria-valuenow={NaN}
              className="MuiLinearProgress-root MuiLinearProgress-colorPrimary test-progress-bar MuiLinearProgress-determinate"
              role="progressbar"
            >
              <div
                className="MuiLinearProgress-bar MuiLinearProgress-barColorPrimary MuiLinearProgress-bar1Determinate"
                style={
                  Object {
                    "transform": "translateX(NaN%)",
                  }
                }
              />
            </div>
          </ForwardRef(LinearProgress)>
        </WithStyles(ForwardRef(LinearProgress))>
      </div>
    </TestProgress>
  </Provider>
</Router>
`;
